src/physics/fermions.jl

```diff
@@ -18,3 +18,338 @@ function parity_sign(P)::Int
   end
   return s
 end
+
+isfermionic(qv::QNVal) = (modulus(qv) < 0)
+
+isfermionic(qn::QN) = any(isfermionic, qn)
+
+has_fermionic_subspaces(i::Index) = false
+
+function has_fermionic_subspaces(i::QNIndex)
+  for b in 1:nblocks(i)
+    isfermionic(qn(i, b)) && (return true)
+  end
+  return false
+end
+
+isfermionic(i::Index) = has_fermionic_subspaces(i)
+
+has_fermionic_subspaces(is::IndexSet) = false
+
+function has_fermionic_subspaces(is::Union{QNIndexSet,NTuple{N,QNIndex}}) where {N}
+  for i in is, b in 1:nblocks(i)
+    isfermionic(qn(i, b)) && (return true)
+  end
+  return false
+end
+
+has_fermionic_subspaces(T) = has_fermionic_subspaces(inds(T))
+
+"""
+    fparity(qn::QN)
+    fparity(qn::IndexVal)
+
+Compute the fermion parity (0 or 1) of a QN of IndexVal,
+defined as the sum mod 2 of each of its fermionic 
+QNVals (QNVals with negative modulus).
+"""
+function fparity(qn::QN)
+  p = 0
+  for qv in qn
+    if isfermionic(qv)
+      p += val(qv)
+    end
+  end
+  return mod(p, 2)
+end
+
+fparity(iv::IndexVal) = fparity(qn(iv))
+
+Base.isodd(q::QN) = isodd(fparity(q))
+Base.isodd(iv::IndexVal) = isodd(fparity(iv))
+
+"""
+    compute_permfactor(p,iv_or_qn::Vararg{T,N})
+
+Given a permutation p and a set "s" of QNIndexVals or QNs,
+if the subset of index vals which are fermion-parity
+odd undergo an odd permutation (odd number of swaps)
+according to p, then return -1. Otherwise return +1.
+"""
+
+function compute_permfactor(p, iv_or_qn::Vararg{T,N}; range=1:N) where {T,N}
+  using_auto_fermion() || return 1
+  oddp = @MVector zeros(Int, N)
+  n = 0
+  for j in range
+    if fparity(iv_or_qn[p[j]]) == 1
+      n += 1
+      oddp[n] = p[j]
+    end
+  end
+  return parity_sign(oddp[1:n])
+end
+
+# Default implementation for non-QN IndexVals
+NDTensors.permfactor(p, ivs...) = 1.0
```

Comment:
Probably should return `1` instead of `1.0` to match the QN version. 

Maybe in general it should accept a type parameter:
```julia
NDTensors.permfactor(T, p, ivs...) = one(T)
```
since `1` doesn't always preserve the type:
```
julia> typeof(one(Int32) * 1)
Int64
```
in the case that an ITensor has type `Int32`.

Reply:
Good catch